### PR TITLE
[Release] Migrate (almost all) remaining Release Tests to Anyscale Jobs

### PR DESCRIPTION
**Pull request by shomilj**
Signed-off-by: Shomil Jain [shomil@anyscale.com](mailto:shomil@anyscale.com)

Builds on top of https://github.com/ray-project/ray/pull/32135

`staging_v2` and `anyscale_job` are now defaults.

## Tests that have not been migrated

The following tests **have not** been migrated:
- `tune_cloud_gcp_k8s_no_sync_down`
- `tune_cloud_gcp_k8s_ssh_sync`
- `tune_cloud_gcp_k8s_durable_upload`
- `tune_scalability_result_throughput_single_node`
- `long_running_serve_failure`
- `dataset_shuffle_random_shuffle_1tb`
- `chaos_dataset_shuffle_random_shuffle_1tb`
- `chaos_dataset_shuffle_push_based_random_shuffle_1tb`
- `k8s_serve_ha_test`
- `aws_cluster_launcher`

Every other test has been migrated.

Tests that have not been migrated are using `prod_v1` or `staging_v1` envs and `sdk_command` (previous default) or `client` run types.

## Other changes

- `k8s_serve_ha_test` has been modified to fail correctly. Previously it failed while showing up green. The test was not migrated. @sihanwang41 
- `ray-data-bulk-ingest-*` have been fixed to actually run (cluster config was messed up) - they also had disk size increased to 400 GB as `ray-data-bulk-ingest-out-of-core-benchmark` was running out of space @stephanie-wang
- Fixed hard timeouts in `ray_release` not having a time buffer
- Added fail-fast path for fetching jsons in `anyscale_job_runner`

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]

**Comment by AmeerHajAli** [comment]
@Yard1 do we need to wait for this to pass the CI or I could just merge?

File: release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml [hunk 1]
```diff
@@ -6,15 +6,17 @@ max_workers: 0
...
-              DeleteOnTermination: true
 
 worker_node_types:
     - name: worker_node
       instance_type: m5.xlarge
       min_workers: 0
       max_workers: 0
       use_spot: false
+
+aws:
```
Comment (AmeerHajAli) [inline release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml:17]: what is that

File: release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml [hunk 2]
```diff
@@ -6,15 +6,17 @@ max_workers: 0
...
+    BlockDeviceMappings:
+        - DeviceName: /dev/sda1
+          Ebs:
+            DeleteOnTermination: true
+            VolumeSize: 400
```
Comment (AmeerHajAli) [inline release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml:22]: so 300 is not enough?

File: release/k8s_tests/run_gcs_ft_on_k8s.py [hunk 3]
```diff
@@ -401,6 +403,10 @@ def main():
             for p in procs:
                 p.kill()
             print("==== Cleanup done ===")
+
+        if exception:
+            raise exception
```
Comment (AmeerHajAli) [inline release/k8s_tests/run_gcs_ft_on_k8s.py:408]: mmm what is happening here?

File: release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml (hunk 2 above)
Comment (Yard1) [reply release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml:22]: Peak usage was 326 GB - this test was completely broken on master and not running at all

File: release/nightly_tests/dask_on_ray/1tb_sort_compute.yaml [hunk 4]
```diff
@@ -1,8 +1,7 @@
-cloud_id: cld_17WvYIBBkdgLwEUNcLeRAE
+cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
```
Comment (AmeerHajAli) [inline release/nightly_tests/dask_on_ray/1tb_sort_compute.yaml:1]: so how much cheaper is it?

File: release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml (hunk 1 above)
Comment (Yard1) [reply release/air_tests/air_benchmarks/mlperf-train/compute_cpu_16.yaml:17]: Advanced aws config

File: release/ray_release/command_runner/anyscale_job_runner.py [hunk 5]
```diff
@@ -67,6 +67,9 @@ def __init__(
         self.prepare_commands = []
         self._wait_for_nodes_timeout = 0
 
+        self._results_uploaded = True
+        self._metrics_uploaded = True
+
```
Comment (AmeerHajAli) [inline release/ray_release/command_runner/anyscale_job_runner.py:72]: what is this

File: release/k8s_tests/run_gcs_ft_on_k8s.py (hunk 3 above)
Comment (Yard1) [reply release/k8s_tests/run_gcs_ft_on_k8s.py:408]: We reraise the exception after cleanup - this test always exited successfully despite not actually working, as the exception was swallowed

File: release/ray_release/command_runner/anyscale_job_runner.py (hunk 5 above)
Comment (Yard1) [reply release/ray_release/command_runner/anyscale_job_runner.py:72]: Fail fast path - if we know we have not uploaded metrics or results from the cluster to s3, we don't need to retry 3 times when trying to download them

File: release/ray_release/glue.py [hunk 6]
```diff
@@ -210,6 +211,13 @@ def run_release_test(
         cluster_timeout = int(
             test["run"].get("session_timeout", DEFAULT_CLUSTER_TIMEOUT)
         )
+        prepare_cmd = test["run"].get("prepare", None)
+        if prepare_cmd:
+            prepare_timeout = test["run"].get("prepare_timeout", command_timeout)
+        else:
+            prepare_timeout = 0
+        command_and_prepare_timeout = command_timeout + prepare_timeout
+
```
Comment (AmeerHajAli) [inline release/ray_release/glue.py:220]: what is happening here? sorry but it is hard to understand context without a comment

File: release/ray_release/scripts/run_release_test.py [hunk 7]
```diff
@@ -73,7 +74,10 @@
 @click.option(
     "--env",
     default=None,
-    type=click.Choice(["prod", "staging"]),
+    # Get the names without suffixes of all files in "../environments"
+    type=click.Choice(
+        [x.stem for x in (Path(__file__).parent.parent / "environments").glob("*.env")]
+    ),
```
Comment (AmeerHajAli) [inline release/ray_release/scripts/run_release_test.py:80]: why do we need this?

File: release/release_tests.yaml [hunk 8]
```diff
@@ -1506,12 +1416,11 @@
 
   frequency: nightly
   team: ml
-  env: staging
+  env: staging_v1
```
Comment (AmeerHajAli) [inline release/release_tests.yaml:1419]: so this is still on v1?

File: release/release_tests.yaml [hunk 9]
```diff
@@ -4832,6 +4619,8 @@
     script: RAY_DATASET_PUSH_BASED_SHUFFLE=1 python dataset/sort.py --num-partitions=1000 --partition-size=1e9 --shuffle
     wait_for_nodes:
       num_nodes: 20
+    
+    type: sdk_command
```
Comment (AmeerHajAli) [inline release/release_tests.yaml:4623]: why sdk?

File: release/release_tests.yaml [hunk 10]
```diff
@@ -4819,6 +4602,10 @@
 
   stable: false
 
+  # TODO: Migrate this test to Anyscale Jobs / staging_v2 once the test has been confirmed
```
Comment (AmeerHajAli) [inline release/release_tests.yaml:4605]: when?

File: release/release_tests.yaml (hunk 9 above)
Comment (AmeerHajAli) [reply release/release_tests.yaml:4623]: update: saw the comment above.

File: release/release_tests.yaml [hunk 11]
```diff
@@ -4860,6 +4649,7 @@
     test_suite: k8s_tests
 
   stable: false
+  env: prod_v1
```
Comment (AmeerHajAli) [inline release/release_tests.yaml:4652]: what do we plan to do here?

File: release/release_tests.yaml [hunk 12]
```diff
@@ -4871,6 +4661,7 @@
     timeout: 28800 # 8h
     prepare: bash prepare.sh
     script: python run_gcs_ft_on_k8s.py
+    type: sdk_command
```
Comment (AmeerHajAli) [inline release/release_tests.yaml:4664]: why?

File: release/release_tests.yaml (hunk 8 above)
Comment (Yard1) [reply release/release_tests.yaml:1419]: yes, as outlined in description - we'll track those as issues and assign test owners to fix

File: release/release_tests.yaml (hunk 9 above)
Comment (Yard1) [reply release/release_tests.yaml:4623]: this was the old default, so this test has not been migrated

File: release/release_tests.yaml (hunk 12 above)
Comment (Yard1) [reply release/release_tests.yaml:4664]: old default, this test has not been migrated

File: release/ray_release/command_runner/anyscale_job_runner.py (hunk 5 above)
Comment (AmeerHajAli) [reply release/ray_release/command_runner/anyscale_job_runner.py:72]: can you please add a comment?

**Comment by AmeerHajAli** [comment]
@shomilj I think you need to sign off this PR for DCO to pass.

File: release/ray_release/scripts/run_release_test.py (hunk 7 above)
Comment (Yard1) [reply release/ray_release/scripts/run_release_test.py:80]: This allows the script user to change the environment for the test - previously this was hardcoded, now it matches all available environment files

File: release/release_tests.yaml (hunk 11 above)
Comment (Yard1) [reply release/release_tests.yaml:4658]: this test has not been migrated

File: release/k8s_tests/run_gcs_ft_on_k8s.py (hunk 3 above)
Comment (AmeerHajAli) [reply release/k8s_tests/run_gcs_ft_on_k8s.py:408]: Why don't we raise it in row 392?

**Comment by shomilj** [comment]
Signed off!

File: release/k8s_tests/run_gcs_ft_on_k8s.py (hunk 3 above)
Comment (Yard1) [reply release/k8s_tests/run_gcs_ft_on_k8s.py:408]: I didn't want to modify the flow of the script (cleanup) - @sihanwang41 will be rewriting the test anyway

Comment (AmeerHajAli) [reply release/k8s_tests/run_gcs_ft_on_k8s.py:408]: @sihanwang41 , where are we tracking the rewrite?

**Comment by AmeerHajAli** [comment]
Taking a moment of silence to appreciate this moment....

**Pull request merged**